.gitlab-ci.yml

```diff
@@ -0,0 +1,84 @@
+---
+stages:
+  - 'validation'
+
+variables:
+  PUPPET_VERSION:    'UNDEFINED' # <- Matrixed jobs MUST override this (or fail)
+  BUNDLER_VERSION:   '2.2.6'
+
+  # Force dependencies into a path the gitlab-runner user can write to.
+  # (This avoids some failures on Runners with misconfigured ruby environments.)
+  GEM_HOME:          .vendor/gem_install
+  BUNDLE_CACHE_PATH: .vendor/bundle
+  BUNDLE_PATH:       .vendor/bundle
+  BUNDLE_BIN:        .vendor/gem_install/bin
+  BUNDLE_NO_PRUNE:   'true'
+
+  SIMP_SKIP_NON_SIMPOS_TESTS: 1
+
+# bundler dependencies and caching
+#
+# - Cache bundler gems between pipelines foreach Ruby version
+# - Try to use cached and local resources before downloading dependencies
+# --------------------------------------
+.setup_bundler_env: &setup_bundler_env
+  image: 'ruby:2.5' # TODO: Default image for now, should add in images when this repository is ready for matrix testing.
```

Comment:
The jobs in this repository don't require matrix testing, so the repository can safely use a single ruby version to run its rake tasks.  For that reason, this `image:` setting would be best placed as a top-level key.